packages/react-devtools-shared/src/Logger.js

```diff
@@ -40,14 +40,20 @@ export type LogEvent =
     |}
   | {|
       +event_name: 'profiling-start',
+    |}
+  | {|
+      +event_name: 'profiler-tab-changed',
+    |}
+  | {|
+      +event_name: 'settings-changed',
```

Comment:
Do we also want to put the other logs in here?

ex.

```
{|
    +event_name: 'settings-changed',
    +key: string,
    ...
|}
```

Reply:
it's already in metadata :)

Reply:
Hm, I think we should type the metadata too if you'd like to put it in metaData instead. Perhaps

```
{|
  +event_name: 'settings-changed',
  +metadata: {
     +key: string,
      ...
   }
|}
```

Reply:
Metadata is currently a separate thing from the event object.
In my original design metadata is a freeform object that can be used or not used.
But yeah, I agree from this case we can tell it's better if they are defined together so we know for sure some events always have certain metadata attributes. I'll update it.

Reply:
Updated :)